lib/engine-dylib/src/artifact.rs

```diff
@@ -773,4 +793,29 @@ impl Artifact for DylibArtifact {
     fn serialize(&self) -> Result<Vec<u8>, SerializeError> {
         Ok(std::fs::read(&self.dylib_path)?)
     }
+
+    /// Serialize a `DylibArtifact` to a portable file
+    fn serialize_to_file(&self, path: &Path) -> Result<(), SerializeError> {
+        let serialized = self.serialize()?;
+        std::fs::write(&path, serialized)?;
+
+        // Rename .dylib identifier to avoid linking issues
+        if path.extension().unwrap() == "dylib" {
+            let filename = path.file_name().unwrap();
+            let parent_dir = path.parent().unwrap();
+            let absolute_path = std::fs::canonicalize(&parent_dir)
+                .unwrap()
+                .into_os_string()
+                .into_string()
+                .unwrap();
+
+            Command::new("install_name_tool")
+                .arg("-id")
+                .arg(&filename)
+                .arg(&filename)
+                .current_dir(&absolute_path);
+        }
+
+        Ok(())
+    }
```

Comment:
Please can you explain to me why this new function is required :-)?

Reply:
Yes! More specifics can be found on the #wasmer channel on Slack but it was caused by the serialised file referencing the original tempfile created by `DylibArtifact`:

```
nathanhorrigan@Nathans-MBP ~/C/w/examples (master)> otool -L sum.dylib
sum.dylib:
	/var/folders/5_/73q9bnl90d106ddc7smgrq7h0000gn/T/wasmer_dylib_dB8mZ6.dylib (compatibility version 0.0.0, current version 0.0.0)
```

The above renames the identifier in the `.dylib` file to match the new file instead of the old. I am still however having issues getting Xcode to load the `.dylib`...

Reply:
Current Xcode error:
```
dyld: Library not loaded: sum.dylib
  Referenced from: /private/var/containers/Bundle/Application/F45CCF73-A561-4E89-9B58-1AD0693A9A18/Wasm3Example.app/Wasm3Example
  Reason: image not found
dyld: launch, loading dependent libraries
DYLD_LIBRARY_PATH=/usr/lib/system/introspection
DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

Reply:
Please explain this inside the code :-).

Reply:
I've added a description.